src/smc-webapp/course/main.cjsx

```diff
@@ -1223,7 +1225,188 @@ exports.init_redux = init_redux = (redux, course_project_id, course_filename) ->
             # Now open it
             redux.getProjectActions(proj).open_directory(path)
 
+        # Handouts
+        add_handout: (path) =>
```

Comment:
Simplified copies of much of what's happening in similarly named assignment methods. Some cases you could literally change `handout` for `assignment` and you'd have the other function. 

Here there is much more code to potentially reuse but even more so in this case, I would like to not entwine assignments and handouts into the same code.